[BUG] Flaky test: test_commit_authorization_expires_while_parent_pipe_stays_open

### Setup

CI, `uv run pytest -n auto`. Seen on [run 33456471423](https://github.com/stickerdaniel/linkedin-mcp-server/actions/runs/33456471423/job/99697771768), on a PR branch whose diff touches only `scraping/link_metadata.py`, `CONTRIBUTING.md` and `tests/test_link_metadata.py`.

### What Happened

`tests/test_daemon_election.py::TestPublishingLast::test_commit_authorization_expires_while_parent_pipe_stays_open` failed:

```
AssertionError: assert ['probe', 'prepare', 'prepared', 'logged', 'retry', 'control']
                    == ['probe', 'prepare', 'prepared', 'control', 'logged', 'retry']
```

The same test passes locally on `main` and passed on the previous run of the same branch, so this is timing rather than a regression.

`"control"` is appended by `_SuspendedParent.readline` (line 6359), which runs on the reader thread and then blocks on `release.wait()`. `"logged"` and `"retry"` are appended by the main flow. Nothing orders those two threads against each other, so the assertion pins one particular interleaving. Under `-n auto` the reader thread can be scheduled after the warning is logged, and the list comes out in the other order.

Both orderings look equally valid for what the test is actually checking, which is that commit authorization expired while the pipe stayed open.

### Steps to Reproduce

Not reliably reproducible locally; it needs the scheduling pressure of parallel workers. `uv run pytest tests/test_daemon_election.py -k commit_authorization_expires` passes on `main`.

### Logs

```
==== 1 failed, 3040 passed, 57 skipped, 1437 warnings in 143.20s (0:02:23) =====
```

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6353,7 +6353,10 @@ class TestPublishingLast:
     ):
         release = threading.Event()
         control_read = threading.Event()
+        finished = threading.Event()
         order: list[str] = []
+        outcomes: list[int] = []
+        errors: list[BaseException] = []
 
         class _SuspendedParent:
             def readline(self) -> str:
@@ -6368,22 +6371,41 @@ class TestPublishingLast:
             "warning",
             lambda *args, **kwargs: order.append("logged"),
         )
+
+        def run() -> None:
+            try:
+                outcomes.append(
+                    self._run_serve(
+                        tmp_path,
+                        monkeypatch,
+                        order,
+                        control=_SuspendedParent(),
+                    )
+                )
+            except BaseException as exc:  # noqa: BLE001 - asserted by this test
+                errors.append(exc)
+            finally:
+                finished.set()
+
+        child = threading.Thread(target=run, daemon=True)
+        child.start()
         try:
-            outcome = self._run_serve(
-                tmp_path,
-                monkeypatch,
-                order,
-                control=_SuspendedParent(),
+            assert control_read.wait(_BOUNDED_CALL_SECONDS), (
+                "the owner never started reading the open parent pipe"
+            )
+            assert finished.wait(_BOUNDED_CALL_SECONDS), (
+                "the blocked parent pipe outlived the authorization deadline"
             )
         finally:
             release.set()
+            child.join(timeout=5)
 
-        assert control_read.wait(_BOUNDED_CALL_SECONDS)
-        assert outcome == 1
+        assert not child.is_alive()
+        assert errors == []
+        assert outcomes == [1]
         # `control` is appended on the reader thread `_read_control_until` starts.
-        # This parent never returns from `readline`, so the main flow reaches its
-        # deadline on its own and nothing orders the two against each other. Only
-        # the main flow's sequence is part of what this test verifies.
+        # Once that read is known to be blocked, the main flow reaches its deadline
+        # independently. Only the main flow's sequence is ordered across threads.
         assert order.count("control") == 1
         assert [event for event in order if event != "control"] == [
             "probe",

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6381,7 +6381,7 @@ class TestPublishingLast:
         # This parent never returns from `readline`, so the main flow reaches its
         # deadline on its own and nothing orders the two against each other. Only
         # the main flow's sequence is part of what this test verifies.
-        assert "control" in order
+        assert order.count("control") == 1
         assert [event for event in order if event != "control"] == [
             "probe",
             "prepare",

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6370,7 +6370,7 @@ class TestPublishingLast:
             assert not release.is_set()
             order.append("logged")
 
-        monkeypatch.setattr(daemon_owner, "_COMMIT_AUTH_SECONDS", 0.1)
+        monkeypatch.setattr(daemon_owner, "_COMMIT_AUTH_SECONDS", _BOUNDED_CALL_SECONDS)
         monkeypatch.setattr(daemon_owner.logger, "warning", log_timeout)
 
         def run() -> None:

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6370,7 +6370,23 @@ class TestPublishingLast:
             assert not release.is_set()
             order.append("logged")
 
-        monkeypatch.setattr(daemon_owner, "_COMMIT_AUTH_SECONDS", _BOUNDED_CALL_SECONDS)
+        real_thread = threading.Thread
+
+        class _SynchronizedThread(real_thread):
+            def start(self) -> None:
+                real_thread.start(self)
+                if self.name == "daemon-control":
+                    assert control_read.wait(_BOUNDED_CALL_SECONDS), (
+                        "the daemon-control reader never entered the blocking "
+                        "parent-pipe read"
+                    )
+
+        monkeypatch.setattr(daemon_owner, "_COMMIT_AUTH_SECONDS", 0.1)
+        monkeypatch.setattr(
+            daemon_owner,
+            "threading",
+            SimpleNamespace(Thread=_SynchronizedThread),
+        )
         monkeypatch.setattr(daemon_owner.logger, "warning", log_timeout)
 
         def run() -> None:

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6352,11 +6352,13 @@ class TestPublishingLast:
         self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch
     ):
         release = threading.Event()
+        control_read = threading.Event()
         order: list[str] = []
 
         class _SuspendedParent:
             def readline(self) -> str:
                 order.append("control")
+                control_read.set()
                 release.wait()
                 return "commit\n"
 
@@ -6376,6 +6378,7 @@ class TestPublishingLast:
         finally:
             release.set()
 
+        assert control_read.wait(_BOUNDED_CALL_SECONDS)
         assert outcome == 1
         # `control` is appended on the reader thread `_read_control_until` starts.
         # This parent never returns from `readline`, so the main flow reaches its

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6377,11 +6377,15 @@ class TestPublishingLast:
             release.set()
 
         assert outcome == 1
-        assert order == [
+        # `control` is appended on the reader thread `_read_control_until` starts.
+        # This parent never returns from `readline`, so the main flow reaches its
+        # deadline on its own and nothing orders the two against each other. Only
+        # the main flow's sequence is part of what this test verifies.
+        assert "control" in order
+        assert [event for event in order if event != "control"] == [
             "probe",
             "prepare",
             "prepared",
-            "control",
             "logged",
             "retry",
         ]

--- a/tests/test_daemon_election.py
+++ b/tests/test_daemon_election.py
@@ -6365,12 +6365,13 @@ class TestPublishingLast:
                 release.wait()
                 return "commit\n"
 
+        def log_timeout(*args: object, **kwargs: object) -> None:
+            assert control_read.is_set()
+            assert not release.is_set()
+            order.append("logged")
+
         monkeypatch.setattr(daemon_owner, "_COMMIT_AUTH_SECONDS", 0.1)
-        monkeypatch.setattr(
-            daemon_owner.logger,
-            "warning",
-            lambda *args, **kwargs: order.append("logged"),
-        )
+        monkeypatch.setattr(daemon_owner.logger, "warning", log_timeout)
 
         def run() -> None:
             try:

